src/sentry/runner/commands/devservices.py

```diff
@@ -233,10 +232,27 @@ def up(ctx, services, project, exclude, fast, skip_only_if):
 
     get_or_create(ctx.obj["client"], "network", project)
 
-    for name in selected_services:
-        _start_service(
-            ctx.obj["client"], ctx.obj["low_level_client"], name, containers, project, fast=fast
-        )
+    with ThreadPoolExecutor(max_workers=len(selected_services)) as executor:
+        futures = []
+        for name in selected_services:
+            futures.append(
+                executor.submit(
+                    _start_service,
+                    ctx.obj["client"],
+                    ctx.obj["low_level_client"],
+                    name,
+                    containers,
+                    project,
+                    fast=fast,
+                )
+            )
+        for future in as_completed(futures):
+            # future.result() reraises exception if any.
+            # We don't want to raise, just report on it.
```

Comment:
Wouldn't we want it to crash if a container fails to start?

Reply:
Hmm, yeah.

Reply:
Oh, I remember now, the reason why I wrote it this way is that it isn't trivial to crash the entire Python process if an exception happens in a particular future. Like, letting it reraise to the main thread will stop it, but the remaining threads that are executing futures will just continue to run.

Reply:
Oh, I remember now, the reason why I wrote it this way is that it isn't trivial to crash the entire Python process if an exception happens in a particular future. Like, letting it reraise to the main thread will stop it, but the remaining threads that are executing futures will just continue to run.

Reply:
It'd be easy with multiprocessing since parent process can just kill the children. But with threads the only way seems to be having futures check a threadsafe sentinel value periodically, and exit based on that. But there isn't really enough granularity there, you'd likely be waiting for all threads to finish docker pulling anyways...

Reply:
I also tried sending SIGINT noninteractively to the main process since doing it interactively, you have to hit it # times = # threads to stop everything, but it didn't work. lol

Reply:
Oh, I guess, I could just send SIGTERM.